sdk/src/userMap/userMap.ts

```diff
@@ -303,65 +303,59 @@ export class UserMap implements UserMapInterface {
 				},
 			];
 
-			const rpcJSONResponse: any =
-				// @ts-ignore
-				await this.connection._rpcRequest('getProgramAccounts', rpcRequestArgs);
-
-			const rpcResponseAndContext: RpcResponseAndContext<
-				Array<{
-					pubkey: PublicKey;
-					account: {
-						data: [string, string];
-					};
-				}>
-			> = rpcJSONResponse.result;
-
+			// @ts-ignore
+			const rpcJSONResponse: any = await this.connection._rpcRequest('getProgramAccounts', rpcRequestArgs);
+			const rpcResponseAndContext: RpcResponseAndContext<Array<{ pubkey: PublicKey; account: { data: [string, string]; } }>> = rpcJSONResponse.result;
 			const slot = rpcResponseAndContext.context.slot;
-
 			this.updateLatestSlot(slot);
 
 			const programAccountBufferMap = new Map<string, Buffer>();
-			for (const programAccount of rpcResponseAndContext.value) {
+			rpcResponseAndContext.value.forEach(programAccount => {
 				programAccountBufferMap.set(
 					programAccount.pubkey.toString(),
 					// @ts-ignore
-					Buffer.from(
-						programAccount.account.data[0],
-						programAccount.account.data[1]
-					)
+					Buffer.from(programAccount.account.data[0], programAccount.account.data[1])
 				);
-			}
+			});
+
+			const concurrencyLimit = 20;
+			const semaphore = new Array(concurrencyLimit).fill(Promise.resolve());
```

Comment:
what's the reason for this semaphore? does it still work if you just let all the promises run at once?